src/services/fetchers/ByBitFetcher.ts

```diff
@@ -0,0 +1,103 @@
+import axios from 'axios';
+import {inject, injectable} from 'inversify';
+import {Logger} from 'winston';
+
+import Settings from '../../types/Settings.js';
+
+type FeedValue = {fsym: string; tsym: string};
+
+export type InputParams = Map<string, FeedValue>;
+
+export interface OutputValue {
+  fsym: string;
+  tsym: string;
+  value: number;
+}
+
+@injectable()
+class ByBitFetcher {
+  @inject('Logger') protected logger!: Logger;
+
+  private timeout: number;
+  private inputs: InputParams;
+  private usdtCurrencyRegex: RegExp;
+
+  constructor(@inject('Settings') settings: Settings) {
+    this.timeout = settings.api.byBit.timeout;
+    this.inputs = new Map();
+    this.usdtCurrencyRegex = /USDT$/;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types,@typescript-eslint/no-explicit-any
+  async apply(inputs: InputParams): Promise<OutputValue[]> {
+    this.inputs = inputs;
+    const baseURL = process.env.NODE_ENV !== 'testing' ? 'https://api.bybit.com' : 'https://api-testnet.bybit.com';
+    const sourceUrl = `${baseURL}/v5/market/tickers?category=spot`;
+
+    this.logger.debug(`[ByBitFetcher] call for: ${baseURL}`);
+
+    const response = await axios.get(sourceUrl, {
+      timeout: this.timeout,
+      timeoutErrorMessage: `Timeout exceeded: ${sourceUrl}`,
+    });
+
+    if (response.status !== 200) {
+      throw new Error(response.data);
+    }
+
+    if (response.data.Response === 'Error') {
+      throw new Error(response.data.Message);
+    }
+
+    const feedValues = this.resolveFeeds(inputs, response.data.result.list);
+
+    return feedValues ?? [];
+  }
+
+  private resolveFeeds(inputs: InputParams, priceList: Record<string, string>[]): OutputValue[] {
+    const outputs: OutputValue[] = [];
+    const priceDatas = priceList.filter((entry: Record<string, string>) => this.filterFeeds(entry));
+
+    priceDatas.forEach((priceData: Record<string, string>) => {
+      const feed = this.getInputFeed(priceData)!;
+
+      if (this.usdtCurrencyRegex.test(priceData.symbol)) {
+        // Some pairs doesn't have usdIndexPrice. e.g.: TAMAUSDT
+        if (!priceData.usdIndexPrice) {
+          this.logger.debug(`[ByBitFetcher] ${priceData.symbol}: No usd price`);
+          return;
+        }
+
+        this.logger.debug(`[ByBitFetcher] resolved price: ${priceData.symbol}: ${priceData.usdIndexPrice}`);
+
+        outputs.push({tsym: feed.tsym, fsym: feed.fsym, value: Number(priceData.usdIndexPrice)});
+        return;
+      }
+
+      this.logger.debug(`[ByBitFetcher] resolved price: ${priceData.symbol}: ${priceData.lastPrice}`);
+
+      outputs.push({
+        fsym: feed?.fsym,
```

Comment:
if feed is empty, we should log error and not push empty date into output result.

Reply:
Done